### PR TITLE
Slimmer, .(git|docker)ignore, config files in files not shell scripts

### PR DESCRIPTION
**Pull request by sam-qordoba**
This PR:

* uses debian-slim, resulting in an image that is more than 50% smaller
* adds .gitignore and .dockerignore files
* makes `setup.sh` unnecessary by creating `.streamlit/config.toml` and `.streamlit/credentials.toml` and copying them into the docker image. It is still hard to change the port streamlit runs on inside docker and expose it, but it is easier than it was, and it is documented in README

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Editors
```
Comment (themightychris) [inline .gitignore]: please drop this giant grab-bag of ignore rules, there's lots of really generic names and things we might want to version in here. A minimal one was added in #24, let's add to that as needed. If anyone accidentally commits extra junk we'll catch it in PR review

Comment (sam-qordoba) [reply .gitignore]: Can I keep things like `.vscode` and `venv`? I have those locally and don't want them showing up in the git history

Comment (themightychris) [reply .gitignore]: @sam-qordoba I plan on committing some `.vscode/` eventually, its contents are intended to travel with the project to give devs an environment. If you have useful .vscode configuration please consider contributing (maybe in another PR)

It sounds like putting `venv/` into `.gitignore` is a best practice, as other devs following our official workflow will end up with one and we don't want them committing it

FYI, there is a `.git/info/exclude` file which is a local-only .gitignore, it works the same way and you can use it for masking out

Comment (sam-qordoba) [reply .gitignore]: So is this smaller version okay for now? And when you want to add in a `.vscode/` you can remove it from here?

**Pull request merged**